### PR TITLE
[Host.Outbox.Sql] Abort publishing of unknown message types

### PR DESCRIPTION
**Pull request by EtherZa**
`OutboxSendingTask` will immediately abort sending a message when the `MessageType` fails to resolve.

Changes:
* `MessageType` resolution moved out of `SqlOutboxRepository` and into `OutboxSendTask` and `OutboxForwardingPublishInterceptor`. 
* When `OutboxSendTask` fails to resolve the 'MessageType', an error is logged and the message is flagged as 'DeliveryAborted'.
* When `OutboxSendTask` fails to resolve the associated bus. The affected messages are also immediately aborted (previously, n attempts would be made before doing the same).

### CODE REVIEW

File: src/SlimMessageBus.Host.Outbox/Services/OutboxSendingTask.cs [hunk 1]
```diff
@@ -242,67 +239,94 @@ internal async Task<int> SendMessages(IServiceProvider serviceProvider, IOutboxR
...
-                        {
-                            _logger.LogWarning("Not able to find matching bus provider for the outbox message with Id {MessageId} of type {MessageType} to pathGroup {Path} using {BusName} bus. The message will be skipped.", outboxMessage.Id, outboxMessage.MessageType.Name, outboxMessage.Path, outboxMessage.BusName);
-                        }
-                        else
-                        {
-                            _logger.LogWarning("Bus provider for the outbox message with Id {MessageId} of type {MessageType} to pathGroup {Path} using {BusName} bus does not support bulk processing. The message will be skipped.", outboxMessage.Id, outboxMessage.MessageType.Name, outboxMessage.Path, outboxMessage.BusName);
-                        }
-
-                        ids.Add(outboxMessage.Id);
+                        _logger.LogWarning("Not able to find matching bus provider for the outbox message with Id {MessageId} of type {MessageType} to pathGroup {Path} using {BusName} bus. The message will be skipped.", outboxMessage.Id, outboxMessage.MessageType, outboxMessage.Path, outboxMessage.BusName);
```
Comment (zarusz) [inline src/SlimMessageBus.Host.Outbox/Services/OutboxSendingTask.cs:273]: `to path` (remove `Group`)

File: src/SlimMessageBus.Host.Outbox/Services/OutboxSendingTask.cs [hunk 2]
```diff
@@ -242,67 +239,94 @@ internal async Task<int> SendMessages(IServiceProvider serviceProvider, IOutboxR
...
+                    }
+                    else
+                    {
+                        _logger.LogWarning("Bus provider for the outbox message with Id {MessageId} of type {MessageType} to pathGroup {Path} using {BusName} bus does not support bulk processing. The message will be skipped.", outboxMessage.Id, outboxMessage.MessageType, outboxMessage.Path, outboxMessage.BusName);
```
Comment (zarusz) [inline src/SlimMessageBus.Host.Outbox/Services/OutboxSendingTask.cs:277]: Same here - remove `Group`

File: src/SlimMessageBus.Host.Outbox/Interceptors/OutboxForwardingPublishInterceptor.cs [hunk 3]
```diff
@@ -71,12 +73,12 @@ public async Task OnHandle(T message, Func<Task> next, IProducerContext context)
             BusName = busMaster.Name,
             Headers = context.Headers,
             Path = context.Path,
-            MessageType = messageType,
+            MessageType = this._outboxSettings.MessageTypeResolver.ToName(messageType),
```
Comment (zarusz) [inline src/SlimMessageBus.Host.Outbox/Interceptors/OutboxForwardingPublishInterceptor.cs:76]: `this.` is redundant here

**Comment by zarusz** [comment]
@EtherZa 2 minor comments added. 

Could you please add a note about how a missing message type will be handled to the outbox docs?

Then I am good to merge.

Thanks for your contribution.

**Comment by zarusz** [comment]
@EtherZa any chance you could increase the coverage on SqlOutboxRepository around the Abort methods and remove the unused variable flagged by Sonar?

**Comment by EtherZa** [comment]
@zarusz I have resolved the Sonar complaint and pushed the boundary again to get the requested coverage on `SqlOutboxRepository` by introducing [TestContainers](https://testcontainers.com/) -I am assuming that you are keen to move in this direction as per your comment in [277](https://github.com/zarusz/SlimMessageBus/issues/277#issuecomment-2212584931).

Please let me know if you are happy with the approach and I will squash/sign the commit.

**Comment by zarusz** [comment]
@EtherZa, I appreciate you pushing this over the finish line despite project deadlines!

Yes, happy to merge. Please squash and I am going to release a minor release soon.

Thanks!

**Comment by zarusz** [comment]
Can you also resolve the conflict, I just merged another PR @EtherZa ?

**Comment by EtherZa** [comment]
> @EtherZa, I appreciate you pushing this over the finish line despite project deadlines!
> 
> Yes, happy to merge. Please squash and I am going to release a minor release soon.
> 
> Thanks!

No worries. I've rebased and pushed. All should be good to go now :)

**Pull request merged**